File: Core/Domain/Models/Hospital.cs
namespace Domain.Models
{
    public class Hospital : BaseEntity<int>
    {
        public string Name { get; set; } = default!;
        public string Address { get; set; } = default!;
        public string ContactPhone { get; set; } = default!;
        public decimal Latitude { get; set; }
        public decimal Longitude { get; set; }
        public int AvailableBeds { get; set; }
        public int BedCapacity { get; set; }
        public int AvailableICU { get; set; }
        public int ICUCapacity { get; set; }
        public Shared.Enums.HospitalStatus Status { get; set; }
        public bool IsDeleted { get; set; }
        public decimal StartingPrice { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public bool IsAvailable { get; set; } = true;
        public decimal BasePrice { get; set; }
        public int TotalRequests { get; set; }
        public int AcceptedRequests { get; set; }

        // Navigation Properties
        public ICollection<Assignment> Assignments { get; set; } = [];
        public ICollection<TripReport> TripReports { get; set; } = [];
        public ICollection<Feedback> Feedbacks { get; set; } = [];
        public ICollection<Request> Requests { get; set; } = [];
        public ICollection<ApplicationUser> Staff { get; set; } = [];
    }
}

File: mobile/rescufy/lib/domain/entities/hospital.dart
import 'package:equatable/equatable.dart';

class Hospital extends Equatable {
  const Hospital({
    required this.id,
    required this.name,
    required this.address,
    required this.contactPhone,
    required this.latitude,
    required this.longitude,
    required this.availableBeds,
    required this.bedCapacity,
    required this.availableIcu,
    required this.icuCapacity,
    required this.status,
    required this.isAvailable,
    required this.startingPrice,
    this.distanceKm,
  });

  final int id;
  final String name;
  final String address;
  final String contactPhone;
  final double latitude;
  final double longitude;
  final int availableBeds;
  final int bedCapacity;
  final int availableIcu;
  final int icuCapacity;
  final String status;
  final bool isAvailable;
  final double startingPrice;
  final double? distanceKm;

  Hospital copyWith({double? distanceKm}) {
    return Hospital(
      id: id,
      name: name,
      address: address,
      contactPhone: contactPhone,
      latitude: latitude,
      longitude: longitude,
      availableBeds: availableBeds,
      bedCapacity: bedCapacity,
      availableIcu: availableIcu,
      icuCapacity: icuCapacity,
      status: status,
      isAvailable: isAvailable,
      startingPrice: startingPrice,
      distanceKm: distanceKm ?? this.distanceKm,
    );
  }

  String get directionsUrl =>
      'https://www.google.com/maps/dir/?api=1&destination=$latitude,$longitude';

  String get phoneUrl => 'tel:$contactPhone';

  @override
  List<Object?> get props => [
    id,
    name,
    address,
    contactPhone,
    latitude,
    longitude,
    availableBeds,
    bedCapacity,
    availableIcu,
    icuCapacity,
    status,
    isAvailable,
    startingPrice,
    distanceKm,
  ];
}
